### DISCUSSION TITLE
AP does not connect with external power

### DISCUSSION THREAD

**Discussion by fsofras** [micropython/micropython#14355] [ESP32]
when I connect my esp32 to the USB port my boot.py works and I get a succesful connection to my Access point (192.168.4.1). However when I power the module with anexternal 5V power source I can not get the AP working.
Any help?
Thank you

**Comment by peterhinch** [comment]
Firstly your code should be in `main.py` rather than `boot.py`. In general it's best to have your code in a separate file, with `main.py` simply being a single line importing your actual code.

I would debug this by having `main.py` do something really simple like flashing an LED or toggling a pin. Check that it works at the REPL, then check it still works when powering from 5V. If it doesn't, I would suspect a hardware problem. If it does, you can substitute an import of your actual script in the knowledge that `main.py` is actually running.

**Comment by fsofras** [comment]
Thank you Peter
I will follow your advice and come back
Frank

Στις Τρί 23 Απρ 2024, 11:46 ο χρήστης Peter Hinch ***@***.***>
έγραψε:
